### ISSUE TITLE
Website: release url needs updating

### ISSUE THREAD

**Issue by tp-m** [harfbuzz/harfbuzz#2449]
https://www.freedesktop.org/wiki/Software/HarfBuzz/ says tarball releases can be found at https://www.freedesktop.org/software/harfbuzz/release/ but it looks like recent release tarballs have not been uploaded there.

So I think either the website should be changed to point to https://github.com/harfbuzz/harfbuzz/releases or the last few tarballs should be uploaded to the fdo directory as well.

**Comment by khaledhosny** [comment]
Duplicate of 2333

**Comment by khaledhosny** [comment]
Duplicate of #2333

**Comment by tp-m** [comment]
Apologies, I did search the issues list first, but apparently not with the right search term :)

**Comment by tp-m** [comment]
I'm a bit confused by the resolution of #2333 

> We don’t reference freedesktop.org page anymore as well.

What does that mean? I mean, those freedesktop harfbuzz project pages exist, and they do reference fdo locations for tarballs. Should they be removed entirely? harfbuzz.org (first hit in google) redirects to the fdo wiki page though and it clearly isn't up to date with regard to release tarballs? :)

**Comment by khaledhosny** [comment]
We want harfbuzz.org redirected to harfbuzz.github.io eventually (see https://github.com/harfbuzz/harfbuzz/pull/2391#issuecomment-628163135), but AFAIU we don’t have access to the freedesktop page any more to keep it updated.

**Comment by behdad** [comment]
Se we need two / three things:

- Redirect harfbuzz.org to github.  Currently to our main project page.  Eventually if someone bothers to design a new webpage, then we can host that at harfbuzz.github.io.  This was waiting on @behnam; but if I try really hard I can get access to do.

- Put a huge redirect notice on fdo website; ideally move the content there in our git tree or whatever tree, but that's for future as well.  I do have fdo wiki login info it seems, so I can do that.

- Put a warning in the fdo releases/ directory.  Again, I do have my old ssh key lying around.  Just need to find it and log in there and do.  Ideally we can keep mirroring releases there for a while longer.

So, yes.  All I can do.  Just annoying.  I'm sorry in advance.

Alternatively, one of you can file a request with fdo admins to help with some of above.

**Comment by ebraminio** [comment]
> Alternatively, one of you can file a request with fdo admins to help with some of above.

Wanted to go ahead but just lost on their structure, and, feared from the kind of signaling that would have, we are proud that harfbuzz once was a part fdo, we like them to keep us as a listed project I guess but just happened that we aren't using their infrastructure anymore, even thought we can now try their gitlab hosting and so, that's a bit too late for us as many distributions have listed github releases as their updating url.

**Comment by behdad** [comment]
> Wanted to go ahead but just lost on their structure, and, feared from the kind of signaling that would have, we are proud that harfbuzz once was a part fdo and hosted there, we like them to keep us as a listed project I guess but just happened that we aren't using their infrastructure anymore, even thought we can now try their gitlab hosting and so, that's a bit too late for us as many distributions have listed github releases as their updating url.

When FDO was moving to gitlab in Aug 2018 Daniel Stone of fdo admins contacted me and he asked if there's any reason to keep FDO git mirror if we are doing everything on github.  So there won't be any bad blood for sure.  They'd be happy we are taken care of somewhere else.

They are also happy to forward any URLs we might have, for old git repo or releases directory or ...

So yeah, email https://lists.freedesktop.org/mailman/listinfo/sitewranglers or something, CC me, request what you want.

The only think we still using there is the mailing list.  Again, as I've signaled multiple times over the year, I find it hard to use and I prefer github issues.  We can keep the list dormant and not advertise it much if at all...

**Comment by ebraminio** [comment]
Just asked from sitewranglers, let's see what happens.

**Comment by tp-m** [comment]
I was going to say best to file an issue here: https://gitlab.freedesktop.org/freedesktop/freedesktop/-/issues (rather than the mailing list), but I guess we can wait and see :)

I'm sure whatever you want can be sorted out, whether that's keeping the web page there and restoring some access or adding a permanent redirect to somewhere else.

**Comment by ebraminio** [comment]
Oh, thanks, posted the same here https://gitlab.freedesktop.org/freedesktop/freedesktop/-/issues/271 Thanks!

**Comment by behdad** [comment]
Okay thanks.

Can we agree what we want as our landing page though?  harfbuzz.github.io currently isn't the most inviting landing page.  It's our docs page.  I was suggesting just redirecting to main github project page.

**Comment by ebraminio** [comment]
Our README.md is also isn't in a better shape. GitHub project usually do their at best at README.md, do you think it would be possible to merge https://github.com/harfbuzz/harfbuzz/wiki and https://www.freedesktop.org/wiki/Software/HarfBuzz/ into either harfbuzz.github.io or README.md?

**Comment by behdad** [comment]
I think the github landing page is best.  That already have badges.  A link or two to docs, and one to wiki if we ever populate.  History can also go into a separate .md file.

The points being:
1. This way the download tarball also has all the relevant links.
2. Our target is developers.  Github is familiar to them.

**Comment by behdad** [comment]
Bonus: we don't have to explain where to report bugs, where to download, etc.

**Comment by behdad** [comment]
Okay I put a big shout message on https://www.freedesktop.org/wiki/Software/HarfBuzz/

That's good enough for immediate need.  I've asked Behnam for redirectly URL as well.  I think we don't need sitewranglers right now.  I'll comment there.

**Comment by behdad** [comment]
Okay I also put a note in the old release directory.

While there, I also downloaded newer releases from github to fdo.  While doing that I noticed two changes to how Ebrahim is doing releases compared to me:

1. Instead of "harfbuzz-2.6.5-win32.zip" we have "harfbuzz-win32.zip".  Same for 2.6.6 and 2.6.7.  I have no idea how this happened since the the win32 tarball is also autogenerated.  So I'm not sure what happened. :(  Please rename those and make sure doesn't happen in the future.  The directory inside those might also need renaming.

2. The .sha* checksum and its .asc PGP-signed files were necessary in the old days of FTP and HTTP.  I deem them unnecessary for github since uses HTTPS so there's a established chain of trust already.  We must still sign the tags.  But I think these two are unnecessary and no one uses them.  So feel free to rip off the code and remove or whatever you want.

**Comment by behdad** [comment]
Yoohoo.  Domain fwded now too.  Maybe we add a link to the old fdo page from our readme or wiki such that that page is still reachable until we move that info here.

**Comment by behnam** [comment]
Update: Just set up HTTP Redirect (302) for harfbuzz.org to https://github.com/harfbuzz/harfbuzz/ .

**Comment by behnam** [comment]
Btw, https://github.com/harfbuzz/harfbuzz/blob/master/README.md now needs update re this:

> For bug reports, mailing list, and other information please visit:
> http://harfbuzz.org/

**Comment by behdad** [comment]
> For bug reports, mailing list, and other information please visit:
> http://harfbuzz.org/

Or we put that info in our README.md.

Since domain names are irrelevant these days, I like to be able to retire the domain name in a few years.  Or maybe not.  But the fact is people will link to https://github.com/harfbuzz/harfbuzz instead of harfbuzz.org, because 1) if you open harfbuzz.org that's just a redirect, and 2) people are familiar with github.

So, next question is, do we use harfbuzz.org in our readme and documentation, or the github URL?

Or maybe we actually find someone to finish the docs / wiki / etc merge into a website that goes to github.io and we actually host the domain there.

Ah, back to that idea of raising money or getting new contributors to help with.   Let me put them on twitter!

**Comment by ebraminio** [comment]
> Instead of "harfbuzz-2.6.5-win32.zip" we have "harfbuzz-win32.zip". Same for 2.6.6 and 2.6.7. I have no idea how this happened since the the win32 tarball is also autogenerated. So I'm not sure what happened. :( Please rename those and make sure doesn't happen in the future. The directory inside those might also need renaming.

This has something to do with the I was getting them which is improved now.

> The .sha* checksum and its .asc PGP-signed files were necessary in the old days of FTP and HTTP. I deem them unnecessary for github since uses HTTPS so there's a established chain of trust already. We must still sign the tags. But I think these two are unnecessary and no one uses them. So feel free to rip off the code and remove or whatever you want.

No mention of this implicit knowledge on RELEASING as I just tried to follow that, however, all the better! https://github.com/harfbuzz/harfbuzz/pull/2451

**Comment by behdad** [comment]
Thank you @ebraminio.
